### PR TITLE
MRG: Fix viz stuff (minor)

### LINKED ISSUE #5060: Topomap colorbars
As @agramfort has pointed out, colorbars on topomaps can be too wide:

https://7444-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.krdig2e/html/auto_tutorials/plot_visualize_evoked.html#sphx-glr-auto-tutorials-plot-visualize-evoked-py

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Two small-ish viz bugs in joint plots.

Closes #5060

### CODE REVIEW

File: mne/viz/utils.py [hunk 1]
```diff
@@ -1662,7 +1662,7 @@ def _prepare_joint_axes(n_maps, figsize=None):
     map_ax = [plt.subplot(4, ts, x + 2 + ts) for x in range(n_maps)]
     # Position topomap subplots on the second row, starting on the
     # second column
-    cbar_ax = plt.subplot(4, 3 * (ts + 1), 6 * (ts + 1))
+    cbar_ax = plt.subplot(4, 5 * (ts + 1), 10 * (ts + 1))
```
Comment (jona-sassenhagen) [inline mne/viz/utils.py]: Slimmer cbar

File: mne/viz/utils.py [hunk 2]
```diff
@@ -2131,7 +2131,8 @@ def _setup_butterfly(params):
     params['plot_fun']()
 
 
-def _connection_line(x, fig, sourceax, targetax, y=1):
+def _connection_line(x, fig, sourceax, targetax, y=1.,
+                     y_source_transform="transAxes"):
```
Comment (jona-sassenhagen) [inline mne/viz/utils.py]: TFR plots need data, joint plots need axes coordinates.

**Comment by jona-sassenhagen** [comment]
Waiting for Circle ...

**Comment by jona-sassenhagen** [comment]
Circle looks good:

https://7514-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_visualize_evoked.html

https://7514-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_sensors_time_frequency.html

**Comment by jona-sassenhagen** [comment]
@cbrnr can you do a quick review?

**Comment by agramfort** [comment]
this image https://7514-1301584-gh.circle-artifacts.com/0/html/_images/sphx_glr_plot_visualize_evoked_006.png in 
https://7514-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_visualize_evoked.html is still problematic.

**Comment by larsoner** [comment]
@agramfort that is by design, the user has passed the axes and created them in a naive way (equal sizes)

**Comment by agramfort** [comment]
ok to merge then :)

**Comment by jona-sassenhagen** [comment]
@mmagnuski could also review + merge (it's very short)

**Comment by cbrnr** [comment]
Should we not suggest a better (less naive) way of creating a new axes for the colorbar? Otherwise people might just copy and paste (e.g. by passing a `gridspec_kw` argument which sets `'width_ratios'` accordingly (see [here](https://martinos.org/mne/stable/auto_examples/time_frequency/plot_time_frequency_erds.html#sphx-glr-auto-examples-time-frequency-plot-time-frequency-erds-py)). Other than that LGTM.

**Comment by jona-sassenhagen** [comment]
I think the idea is to keep things very stupid here.

**Comment by cbrnr** [comment]
Why? I don't like it.

**Comment by jona-sassenhagen** [comment]
@cbrnr see https://github.com/mne-tools/mne-python/pull/3528

**Comment by mmagnuski** [comment]
Looks good, I don't see any problem in setting the colorbar this way.

**Comment by jona-sassenhagen** [comment]
Feel free to hit that green button!

**Comment by mmagnuski** [comment]
When you use your phone only the comment button is green :)

**Comment by jona-sassenhagen** [comment]
Thanks @mmagnuski , @cbrnr , @agramfort , @larsoner

**Comment by mmagnuski** [comment]
Thanks @jona-sassenhagen ! 🚀

**Pull request merged**